src/main/java/com/hillel/items_exchange/controller/ImageController.java

```diff
@@ -1,43 +1,99 @@
 package com.hillel.items_exchange.controller;
 
 import com.hillel.items_exchange.dto.ImageDto;
+import com.hillel.items_exchange.model.BaseEntity;
+import com.hillel.items_exchange.model.User;
 import com.hillel.items_exchange.service.ImageService;
+import com.hillel.items_exchange.service.UserService;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
 import javax.validation.constraints.PositiveOrZero;
+import javax.validation.constraints.Size;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Collection;
 import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static org.springframework.http.MediaType.*;
 
 @RestController
 @RequestMapping("/image")
 @RequiredArgsConstructor
 @Validated
 @Slf4j
 public class ImageController {
+    private static final Set<String> SUPPORTED_TYPES = Set.of(IMAGE_JPEG_VALUE, IMAGE_PNG_VALUE, IMAGE_GIF_VALUE);
     private final ImageService imageService;
+    private final UserService userService;
 
-    @GetMapping("/{product_id}/links")
-    public ResponseEntity<List<String>> getImageLinksByProductId(@PathVariable("product_id")
-                                                                @PositiveOrZero(message = "{invalid.id}") long id) {
-        List<String> links = imageService.getLinksByProductId(id);
-        return (links.isEmpty())
+    @GetMapping(value = "/{product_id}/resource")
+    public ResponseEntity<List<byte[]>> getImageLinksByProductId(@PathVariable("product_id")
```

Comment:
In my humble opinion, it would be nice to rename this controller

Reply:
I think you're saying about the method's name. Yup, I missed it and rename it. Thanks.

Reply:
Yes, about method =)